packages/services/api/src/modules/schema/providers/schema-publisher.ts

```diff
@@ -518,51 +525,7 @@ export class SchemaPublisher {
             safeSchemaChanges: contract.schemaChanges?.safe ?? null,
           })) ?? null,
       });
-    }
-
-    if (checkResult.conclusion === SchemaCheckConclusion.Success) {
-      let composition = checkResult.state?.composition ?? null;
-      // in case of a skip this is null
-      if (composition === null) {
-        if (latestVersion == null || latestSchemaVersion == null) {
-          throw new Error(
-            'Composition yielded no composite schema SDL but there is no latest version to fall back to.',
-          );
-        }
-
-        if (latestSchemaVersion.compositeSchemaSDL) {
-          composition = {
-            compositeSchemaSDL: latestSchemaVersion.compositeSchemaSDL,
-            supergraphSDL: latestSchemaVersion.supergraphSDL,
-          };
-        } else {
-          // LEGACY CASE if the schema version record has no sdl
-          // -> we need to do manual composition
```

Comment:
this case did not make sense as schema versions always have SDL if composable. If not composable we should not attempt to compose...

I mixed up schema versions with schema checks when I implemented this as legacy schema check records have no SDL saved.

In addition to that I made the mistake of assuming that `SchemaCheckConclusion.Success` that was skipped is always composable...

Reply:
actually... we have legacy schema version records without SDL, the above still needs to happen in some cases... 😮‍💨

Reply:
handled in https://github.com/kamilkisiela/graphql-hive/pull/3932/commits/396f348fea1aee9cf62eabdb56f626f946db85fd